code/game/objects/structures/candybucket.dm

```diff
@@ -16,16 +16,29 @@
 
 /obj/structure/candybucket/attackby(var/obj/item/I, var/mob/user)
 	..()
+	if(user.isDead() || user.lying || user.incapacitated())
+		return
+	if(!user.Adjacent(src))
+		return
 	if(is_type_in_list(I, allowedTreats) && contents.len < candypacity)
 		if(user.drop_item(I, src))
 			to_chat(user, "<span class='notice'>You add \the [I] to \the [src].</span>")
-			updateBucket()
+			if(istype(I, /obj/item/weapon/reagent_containers/syringe))
+				log_admin("[key_name(user)] added a [I] to [src] at [formatJumpTo(get_turf(usr))], it contains:")
```

Comment:
probably want to send a message_admins with the user when a danger reagent is inserted into the bucket too, so we can easily track griff syringes

Reply:
That one tracks what reagent is in the syringe when added. Figured that would be more useful and fill that purpose.

Reply:
include an actual message_admins though so it's visible when actually actively adminning

Reply:
Oh, I don't know how adminning works. I don't think there's a danger chem list out there so it'll have to annoy admins with every syringe added.

Reply:
You can check if each reagent is in global list `reagents_to_log`

Reply:
I wonder why that's not a var on /datum/reagent

Reply:
also take a look at `log_bad_reagents`